partiql-ast/src/main/java/org/partiql/ast/v1/DataType.java

```diff
@@ -0,0 +1,318 @@
+    }
+
+    public static DataType CHARACTER() {
+        return new DataType(CHARACTER);
+    }
+
+    public static DataType CHARACTER(int length) {
+        return new DataType(CHARACTER, null, null, length);
+    }
+
+    public static DataType VARCHAR() {
+        return new DataType(VARCHAR);
+    }
+
+    public static DataType VARCHAR(int length) {
+        return new DataType(VARCHAR, null, null, length);
+    }
+
+    public static DataType CHARACTER_VARYING() {
+        return new DataType(CHARACTER_VARYING);
+    }
+
+    public static DataType CHARACTER_VARYING(int length) {
+        return new DataType(CHARACTER_VARYING, null, null, length);
+    }
+
+    public static DataType STRING() {
+        return new DataType(STRING);
+    }
+
+    public static DataType STRING(int length) {
+        return new DataType(STRING, null, null, length);
+    }
+
+    public static DataType SYMBOL() {
+        return new DataType(SYMBOL);
+    }
+
+    public static DataType BLOB() {
+        return new DataType(BLOB);
+    }
+
+    public static DataType BLOB(int length) {
+        return new DataType(BLOB, null, null, length);
+    }
+
+    public static DataType CLOB() {
+        return new DataType(CLOB);
+    }
+
+    public static DataType CLOB(int length) {
+        return new DataType(CLOB, null, null, length);
+    }
+
+    public static DataType DATE() {
+        return new DataType(DATE);
+    }
+
+    public static DataType STRUCT() {
+        return new DataType(STRUCT);
+    }
+
+    public static DataType TUPLE() {
+        return new DataType(TUPLE);
+    }
+
+    public static DataType LIST() {
+        return new DataType(LIST);
+    }
+
+    public static DataType SEXP() {
+        return new DataType(SEXP);
+    }
+
+    public static DataType BAG() {
+        return new DataType(BAG);
+    }
+
+    public static DataType TIME() {
+        return new DataType(TIME);
+    }
+
+    public static DataType TIME(int precision) {
+        return new DataType(TIME, precision, null, null);
+    }
+
+    public static DataType TIMESTAMP() {
+        return new DataType(TIMESTAMP);
+    }
+
+    public static DataType TIMESTAMP(int precision) {
+        return new DataType(TIMESTAMP, precision, null, null);
+    }
+
+    public static DataType TIME_WITH_TIME_ZONE() {
+        return new DataType(TIME_WITH_TIME_ZONE);
+    }
+
+    public static DataType TIME_WITH_TIME_ZONE(int precision) {
+        return new DataType(TIME_WITH_TIME_ZONE, precision, null, null);
+    }
+
+    public static DataType TIMESTAMP_WITH_TIME_ZONE() {
+        return new DataType(TIMESTAMP_WITH_TIME_ZONE);
+    }
+
+    public static DataType TIMESTAMP_WITH_TIME_ZONE(int precision) {
+        return new DataType(TIMESTAMP_WITH_TIME_ZONE, precision, null, null);
+    }
+
+    public static DataType INTERVAL() {
+        return new DataType(INTERVAL);
+    }
+
+    public static DataType USER_DEFINED() {
+        return new DataType(USER_DEFINED);
+    }
```

Comment:
We need the identifier of a user-defined type. The code is perfect, but there should be a `private final Identifier name` which is only NON-null for the UDT

Reply:
Ah I missed that. I'll add a new private constructor for just the user-defined type `name`.